src/components/JustNotSorry.js

```diff
@@ -3,160 +3,105 @@ import ReactDOM from 'react-dom';
 
 import Warning from './Warning.js';
 import * as Util from './util.js';
-import WARNING_MESSAGES from '../warnings/phrases.json';
+import WARNINGS from '../warnings/phrases.json';
 import domRegexpMatch from 'dom-regexp-match';
+import { handleContentEditableContentInsert } from '../callbacks/ContentEditableInsert';
+import { handleContentEditableChange } from '../callbacks/ContentEditableChange';
 
-export const WAIT_TIME_BEFORE_RECALC_WARNINGS = 500;
+const WAIT_TIME_BEFORE_RECALC_WARNINGS = 1;
```

Comment:
With this change, I'm seeing performance issues for long emails.  To reproduce, try pasting the following into a compose window 20 times:
> I'm sorry about being so just sorry. Here we go. Really!!! Does that make sense? It actually works!!! I'm sorry.  I'm just so sorry.  Please.  I'm actually just sorry about being so sorry.

Profiling it shows that `getClientRects` is triggering recalculate style over and over again.
![image](https://user-images.githubusercontent.com/1520763/107239744-c8fba500-69ee-11eb-9477-520fdffa9a33.png)
The easy fix is to increase the wait time again.  If you have ideas for how to improve the performance without a debounce, that would be even better (probably a separate PR, though).

Reply:
Changes put back and is working again. One performance gain would be to remove the blur event listener. Is there a reason we remove the highlights when the user clicks out from the message?

It might be more consistent with how spell checks remain highlighted when a user shifts focus away from an email?

Reply:
Removing warnings on blur has been there since the very early days of the extension.  I can't quite remember why.  I'll have to check with Tami, the PM/project founder.

I hate how we remove and re-add all the warnings every time a character is typed (I put that in there, so I can complain with a clear conscience).  I doubt that Grammerly does the same thing.  I struggled with figuring out an easy way to keep track of the warnings and still be able to handle edits to existing text. Remove & re-add was a quick fix that works, but it's terrible for performance.

Reply:
I did a bit more testing.  You're right, the on blur seems to have a big impact.  I'm still seeing performance issues on really long emails, though, even with the delay moved back up to 500. 

Looks like 2.0.0-beta.4 has performance issues, too, so it's not this PR.

1.6.3 is able to handle really long emails with no problem, though.  

We'll have to fix the performance issues on the 2.X branch before this can be released. I'll create a separate GitHub issue to track it.

Reply:
I added #110